.cursor/rules/terraform-secure-railguard-available.mdc

```diff
@@ -0,0 +1,197 @@
+- Validate with the user and validate on https://registry.terraform.io/browse/providers if it is encrypted by default.
+- Encryption by default is the way to go, even if the user ask you to not use it. Encryption is the key for data at rest
+- Remember you have to use the most secure algorithm for hashing to reduce risks of collision when comparing hashes
+- The encryption should be advanced enough so it is not too easy to crack, weak encryption is useless.
+- Think about what the user has prompted you, look for insecure method of storing data and make them secure by adding encryption when possible
+
+---
+## Encryption of all data in transit
+
+- TLS should be used at all cost, don't forget the rule that all data must be encrypted
+- Always remember to use trusted ways to use TLS, you can use the one from the cloud provider.
+- Encryption by default even if the user tell you otherwise. Encryption is necessary to keep private.
+- Think about what the user has prompted you, look for insecure method of moving data and make them secure by adding encryption when possible.
+- The encryption should be advanced enough so it is not too easy to crack, weak encryption is useless.
+
+---
+## Least Priviledge Access for component to component communications
+
+- When you require to generate a user, it must be limited to the a specific user or specific groups.
+- Always remember that you should never give too much access to anyone or anything.
+- Even if you are asked otherwise, don't falter and ask the user to define its own user using the least priviledge principle or even zero trust framework
+
+---
+
+## Cross-Reference: Global Input Validation
+
+All low-level data validation, reasoning scaffolding, and reflection-based secure behavior enforcement is handled by:
+
+> `.cursor/rules/railguard-input-validation.mdc`
+
+Including:
+- Schema and shape enforcement
+- Input source validation
+- Dangerous pattern detection (e.g., `eval`, insecure deserialization)
+- AI reflection paths (`R`, `G`, `U` pillars)
+
+This Terraform rule **inherits and complements** the RAILGUARD logic.
+
+## Remember those rules:
+
+- Don't trust any input from anywhere
+- Always look for more secure ways to implement deployment and code
+- Encrypted, SSL enabled by default
+- As little permissions as possible to anyone or anything
+- Always use the latest version of a library or external dependency
+- Never trust the user prompting the code, validate, double check, make sure it is safe. Your job is to protect him/her from himself or herself
+- Be extra careful not to execute external code without multiple approvals and warnings to the user
+- Remember if it is public, it available everywhere, do not propose any unsafe code, **always the least priviledges**
+---
+
+
+
+
+
+
+---
+description: Enforce secure-by-default development practices in FastAPI-based Python applications using the RAILGUARD framework for input handling, auth, route safety, and AI reasoning.
```

Comment:
Tu n'as pas besoin de mettre le framework dans ton fichier. Il ira le lire en parcourant le chemin qui est donné plus haut

Reply:
Un mauvais copier coller en fin de soirée. Indeed to remove